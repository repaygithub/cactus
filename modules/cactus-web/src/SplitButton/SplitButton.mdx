---
name: SplitButton
menu: Components
---

import PropsTable from 'website-src/components/PropsTable'
import SplitButton from './SplitButton'
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live'

# SplitButton

The `SplitButton` component is a combination of a button and a dropdown and it is useful for providing a user with
some actions to take. It consists of a main action button and any number of other actions that can be displayed by the dropdown.

## Try It Out

export const code = `
<SplitButton onSelectMainAction={() => console.log('Main Action')} mainActionLabel="Main Action">
  <SplitButton.Action onSelect={() => console.log('Action 1')}>Action 1</SplitButton.Action>
  <SplitButton.Action onSelect={() => console.log('Action 2')}>Action 2</SplitButton.Action>
</SplitButton>
`

<LiveProvider code={code} scope={{ SplitButton }}>
  <LiveEditor style={livePreviewStyle} />
  <LiveError />
  <LivePreview />
</LiveProvider>

## Best practices

- The `SplitButton` should be used when there are many distinct possible actions, but only one primary action which
is likely to be used the most, as opposed to the `MenuButton`, which is used to offer multiple ways of carrying out a
single action.
- Pass `aria-label` to `SplitButton` to provide the dropdown trigger with an accessible label.
- If applicable, pass icons to provide a visual representation of each action to the user.

## Properties

### SplitButton

<PropsTable of={SplitButton} staticProp="SplitButton" />

### SplitButton Action

<PropsTable of={SplitButton} staticProp="SplitButtonAction" />
