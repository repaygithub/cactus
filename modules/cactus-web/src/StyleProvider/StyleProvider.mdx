---
name: StyleProvider
menu: Components
---

import PropsTable from 'website-src/components/PropsTable'
import { livePreviewStyle } from '../helpers/constants'
import StyleProvider from './StyleProvider'
import cactusTheme from '@repay/cactus-theme'

# StyleProvider

The `StyleProvider` is a wrapper for `styled-components`' `ThemeProvider` component which gives you the option to include some global styles.

# Best practices

- Make sure that any components from `@repay/cactus-web` are children of `StyleProvider`. This will ensure that each component has the correct styles.

## Basic usage

```jsx
import React from 'react'
import { StyleProvider } from '@repay/cactus-web'
import cactusTheme from '@repay/cactus-theme'

const App = () => {
  return (
    <StyleProvider theme={cactusTheme} global={true}>
      // Include routes/rest of application here
    </StyleProvider>
  )
}
```

# Breakpoints

The `StyleProvider` has a media queries scale which is based on the provided breaking points and is added to the given theme. The media queries have screen sizes small, medium, large, and extraLarge. The set media queries will be larger than the default, for a mobile first approach.

Props can also vary depending on the current media query. This is done by setting a prop to an array, where the array holds different values for each media query size. The first element of the array maps to the default screen size, and each element after maps to the set media queries from smallest to largest. This is used below in the example code to set the background color.

For more information on Breakpoints and Media Queries visit: https://styled-system.com/theme-specification#breakpoints

## Media Queries Usage

```jsx
import React from 'react'
import { StyleProvider, Box } from '.@repay/cactus-web'
import styled from 'styled-components'

const BreakpointBox = styled(Box)`
  border-radius: 15%;

  ${p => p.theme.mediaQueries.small} {
    border-radius: 25%;
  }
  ${p => p.theme.mediaQueries.medium} {
    border-radius: 35%;
  }
  ${p => p.theme.mediaQueries.large} {
    border-radius: 45%;
  }
  ${p => p.theme.mediaQueries.extraLarge} {
    border-radius: 50%;
  }
`
const backgroundColors = ['springgreen', 'mediumturquoise', 'cornflowerblue', 'thistle', 'pink']

const App = () => {
  return (
    <StyleProvider theme={cactusTheme} global={true}>
      <BreakpointBox backgroundColor={backgroundColors} width="100px" height="100px" />
    </StyleProvider>
  )
}
```

# Properties

<PropsTable of={StyleProvider} staticProp="StyleProvider" />
