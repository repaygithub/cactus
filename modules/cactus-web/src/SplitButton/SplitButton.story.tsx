import * as icons from '@repay/cactus-icons/i'
import { boolean, select, text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import React from 'react'

import SplitButton, { IconProps } from './SplitButton'

type IconName = keyof typeof icons | 'None'
const iconNames: IconName[] = Object.keys(icons) as IconName[]

storiesOf('SplitButton', module)
  .add(
    'Basic Usage',
    (): React.ReactElement => {
      iconNames.unshift('None')
      const mainIconName: IconName = select('mainActionIcon', iconNames, 'None')
      const actionIconName1: IconName = select('actionIcon1', iconNames, 'None')
      const actionIconName2: IconName = select('actionIcon2', iconNames, 'None')
      let MainIcon: React.FunctionComponent<IconProps>
      let ActionIcon1: React.FunctionComponent<IconProps>
      let ActionIcon2: React.FunctionComponent<IconProps>
      if (mainIconName !== 'None') {
        MainIcon = icons[mainIconName] as React.FunctionComponent<IconProps>
      }
      if (actionIconName1 !== 'None') {
        ActionIcon1 = icons[actionIconName1] as React.FunctionComponent<IconProps>
      }
      if (actionIconName2 !== 'None') {
        ActionIcon2 = icons[actionIconName2] as React.FunctionComponent<IconProps>
      }
      return (
        <SplitButton
          onSelectMainAction={(): void => {
            console.log('Main Action')
          }}
          mainActionLabel={text('mainActionLabel', 'Main Action')}
          // @ts-ignore
          mainActionIcon={MainIcon ? MainIcon : undefined}
          disabled={boolean('disabled', false)}
        >
          <SplitButton.Action
            onSelect={(): void => console.log('Action One')}
            // @ts-ignore
            icon={ActionIcon1 && ActionIcon1}
          >
            Action One
          </SplitButton.Action>
          <SplitButton.Action
            onSelect={(): void => console.log('Action Two')}
            // @ts-ignore
            icon={ActionIcon2 && ActionIcon2}
          >
            Action Two
          </SplitButton.Action>
        </SplitButton>
      )
    }
  )
  .add(
    'With Collisions',
    (): React.ReactElement => (
      <React.Fragment>
        <div style={{ position: 'absolute', left: '20px', top: '20px' }}>
          Scroll down and to the right
        </div>
        <SplitButton
          onSelectMainAction={(): void => console.log('Main Action')}
          mainActionLabel="Main Action"
        >
          <SplitButton.Action onSelect={(): void => console.log('Action One')}>
            Action One
          </SplitButton.Action>
          <SplitButton.Action onSelect={(): void => console.log('Action Two')}>
            Action Two
          </SplitButton.Action>
        </SplitButton>
      </React.Fragment>
    ),
    { cactus: { overrides: { height: '220vh', width: '220vw' } } }
  )
  .add('Fixed Width Container', () => (
    <div style={{ width: '125px' }}>
      <SplitButton
        onSelectMainAction={(): void => console.log('Main Action')}
        mainActionLabel="Main Action"
      >
        <SplitButton.Action onSelect={(): void => console.log('Action One')}>
          Action One
        </SplitButton.Action>
        <SplitButton.Action onSelect={(): void => console.log('Action Two')}>
          Action Two
        </SplitButton.Action>
      </SplitButton>
    </div>
  ))
