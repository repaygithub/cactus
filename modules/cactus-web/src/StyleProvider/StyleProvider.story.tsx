import React from 'react'

import { storiesOf } from '@storybook/react'
import Box from '../Box/Box'
import cactusTheme from '@repay/cactus-theme'
import Grid from '../Grid/Grid'
import styled from 'styled-components'
import StyleProvider from './StyleProvider'

const backgroundColors = ['springgreen', 'mediumturquoise', 'cornflowerblue', 'thistle', 'pink']

const BreakpointBox = styled(Box)`
  max-width: 320px;
  width: 50px;
  height: 50px;
  border-radius: 15%

  ${p => p.theme.mediaQueries.small} {
    max-width: 708px;
    width: 75px;
    height 75px;
    border-radius: 25%;
  }
  ${p => p.theme.mediaQueries.medium} {
    max-width: 964px;
    width: 100px;
    height: 100px;
    border-radius: 35%;
  }
  ${p => p.theme.mediaQueries.large} {
    max-width: 1140px;
    width: 150px;
    height: 150px;
    border-radius: 45%;
  }
  ${p => p.theme.mediaQueries.extralarge} {
    max-width: 1380px;
    width: 200px;
    height: 200px;
    border-radius: 50%;
  }
`

storiesOf('Style Provider', module).add('Component Adjusments based on Media Queries', () => {
  return (
    <StyleProvider theme={cactusTheme} global={true}>
      <Grid justify="center">
        <Grid.Item tiny={4}>
          <BreakpointBox backgroundColor={backgroundColors} />
        </Grid.Item>
        <Grid.Item tiny={4}>
          <BreakpointBox backgroundColor={backgroundColors} />
        </Grid.Item>
        <Grid.Item tiny={4}>
          <BreakpointBox backgroundColor={backgroundColors} />
        </Grid.Item>
        <Grid.Item tiny={2}>
          <BreakpointBox backgroundColor={backgroundColors} />
        </Grid.Item>
        <Grid.Item tiny={2}>
          <BreakpointBox backgroundColor={backgroundColors} />
        </Grid.Item>
        <Grid.Item tiny={2}>
          <BreakpointBox backgroundColor={backgroundColors} />
        </Grid.Item>
        <Grid.Item tiny={2}>
          <BreakpointBox backgroundColor={backgroundColors} />
        </Grid.Item>
        <Grid.Item tiny={2}>
          <BreakpointBox backgroundColor={backgroundColors} />
        </Grid.Item>
        <Grid.Item tiny={2}>
          <BreakpointBox backgroundColor={backgroundColors} />
        </Grid.Item>
      </Grid>
    </StyleProvider>
  )
})
