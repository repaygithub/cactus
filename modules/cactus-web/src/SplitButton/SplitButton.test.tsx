import { generateTheme } from '@repay/cactus-theme'
import { fireEvent, render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import * as React from 'react'

import animationRender from '../../tests/helpers/animationRender'
import { StyleProvider } from '../StyleProvider/StyleProvider'
import SplitButton from './SplitButton'

describe('component: SplitButton', (): void => {
  describe('mouse interactions', (): void => {
    test('can select main action', (): void => {
      const onMainSelect = jest.fn()
      const { getByText } = render(
        <StyleProvider>
          <SplitButton mainActionLabel="Main Action" onSelectMainAction={onMainSelect}>
            <SplitButton.Action onSelect={jest.fn()}>One Action</SplitButton.Action>
            <SplitButton.Action onSelect={jest.fn()}>Another Action</SplitButton.Action>
          </SplitButton>
        </StyleProvider>
      )

      const mainButton = getByText('Main Action')
      userEvent.click(mainButton)
      expect(onMainSelect).toHaveBeenCalled()
    })

    test('can select an action from the dropdown', async (): Promise<void> => {
      const onAction1Select = jest.fn()
      const { getByText, getByLabelText } = render(
        <StyleProvider>
          <SplitButton mainActionLabel="Main Action" onSelectMainAction={jest.fn()}>
            <SplitButton.Action onSelect={onAction1Select}>One Action</SplitButton.Action>
            <SplitButton.Action onSelect={jest.fn()}>Another Action</SplitButton.Action>
          </SplitButton>
        </StyleProvider>
      )

      const dropdownButton = getByLabelText('Action List') as HTMLButtonElement
      userEvent.click(dropdownButton)
      const action1 = getByText('One Action')
      userEvent.click(action1)
      expect(onAction1Select).toHaveBeenCalled()
    })

    test('cannot select main action when disabled', (): void => {
      const onMainSelect = jest.fn()
      const { getByText } = render(
        <StyleProvider>
          <SplitButton mainActionLabel="Main Action" onSelectMainAction={onMainSelect} disabled>
            <SplitButton.Action onSelect={jest.fn()}>One Action</SplitButton.Action>
            <SplitButton.Action onSelect={jest.fn()}>Another Action</SplitButton.Action>
          </SplitButton>
        </StyleProvider>
      )

      const mainButton = getByText('Main Action')
      userEvent.click(mainButton)
      expect(onMainSelect).not.toHaveBeenCalled()
    })
  })

  describe('keyboard interactions', (): void => {
    test('can select an action from the dropdown', async (): Promise<void> => {
      const onAction2Select = jest.fn()
      const { rerender } = render(
        <StyleProvider>
          <SplitButton mainActionLabel="Main Action" onSelectMainAction={jest.fn()}>
            <SplitButton.Action onSelect={jest.fn()}>One Action</SplitButton.Action>
            <SplitButton.Action onSelect={onAction2Select}>Another Action</SplitButton.Action>
          </SplitButton>
        </StyleProvider>
      )

      const dropdownButton = document.querySelector('[aria-haspopup=menu]') as HTMLButtonElement
      fireEvent.keyDown(dropdownButton, { key: 'Enter' })
      await animationRender()
      // @ts-ignore
      fireEvent.keyDown(document.activeElement, { key: 'ArrowDown' })
      await animationRender()
      rerender(
        <StyleProvider>
          <SplitButton mainActionLabel="Main Action" onSelectMainAction={jest.fn()}>
            <SplitButton.Action onSelect={jest.fn()}>One Action</SplitButton.Action>
            <SplitButton.Action onSelect={onAction2Select}>Another Action</SplitButton.Action>
          </SplitButton>
        </StyleProvider>
      )
      // @ts-ignore
      fireEvent.keyDown(document.activeElement, { key: 'Enter' })
      expect(onAction2Select).toHaveBeenCalled()
    })
  })

  describe('with theme customization', (): void => {
    test('should have square shape', (): void => {
      const theme = generateTheme({ primaryHue: 200, shape: 'square' })
      const { getByText } = render(
        <StyleProvider theme={theme}>
          <SplitButton mainActionLabel="Test" onSelectMainAction={jest.fn()}>
            <SplitButton.Action onSelect={jest.fn()}>One Action</SplitButton.Action>
            <SplitButton.Action onSelect={jest.fn()}>Another Action</SplitButton.Action>
          </SplitButton>
        </StyleProvider>
      )
      const splitButton = getByText('Test')
      const styles = window.getComputedStyle(splitButton)
      expect(styles.borderWidth).toBe('1px')
      expect(styles.borderRadius).toBe('1px')
    })

    test('should have intermediate shape', (): void => {
      const theme = generateTheme({ primaryHue: 200, shape: 'intermediate' })
      const { getByText } = render(
        <StyleProvider theme={theme}>
          <SplitButton mainActionLabel="Test" onSelectMainAction={jest.fn()}>
            <SplitButton.Action onSelect={jest.fn()}>One Action</SplitButton.Action>
            <SplitButton.Action onSelect={jest.fn()}>Another Action</SplitButton.Action>
          </SplitButton>
        </StyleProvider>
      )

      const splitButton = getByText('Test')
      const styles = window.getComputedStyle(splitButton)
      expect(styles.borderWidth).toBe('1px')
      expect(styles.borderRadius).toBe('8px 1px 1px 8px')
    })
    test('dropdown should not have box shadows', (): void => {
      const theme = generateTheme({ primaryHue: 200, boxShadows: false })
      const { getByText } = render(
        <StyleProvider theme={theme}>
          <SplitButton mainActionLabel="Test" onSelectMainAction={jest.fn()}>
            <SplitButton.Action onSelect={jest.fn()}>One Action</SplitButton.Action>
            <SplitButton.Action onSelect={jest.fn()}>Another Action</SplitButton.Action>
          </SplitButton>
        </StyleProvider>
      )

      const splitButton = getByText('Test')
      const styles = window.getComputedStyle(splitButton)
      expect(styles.boxShadow).toBe('')
    })
    test('should have 2px borders', (): void => {
      const theme = generateTheme({ primaryHue: 200, border: 'thick' })
      const { getByText } = render(
        <StyleProvider theme={theme}>
          <SplitButton mainActionLabel="Test" onSelectMainAction={jest.fn()}>
            <SplitButton.Action onSelect={jest.fn()}>One Action</SplitButton.Action>
            <SplitButton.Action onSelect={jest.fn()}>Another Action</SplitButton.Action>
          </SplitButton>
        </StyleProvider>
      )

      const splitButton = getByText('Test')
      const styles = window.getComputedStyle(splitButton)
      expect(styles.borderWidth).toBe('2px')
    })
  })
})
