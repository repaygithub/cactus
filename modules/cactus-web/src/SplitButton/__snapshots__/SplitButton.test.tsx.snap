// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: SplitButton snapshot 1`] = `
.c6 {
  font-size: 8px;
  vertical-align: middle;
}

.c2 {
  box-sizing: border-box;
  border: 1px solid;
  border-radius: 20px 1px 1px 20px;
  background-color: hsl(0,0%,100%);
  height: 32px;
  outline: none;
  font-size: 18px;
  line-height: 28px;
  font-weight: 400px;
  cursor: pointer;
  padding-left: 12px;
  padding-right: 12px;
}

.c2 svg {
  margin-right: 4px;
  margin-bottom: 3px;
}

.c2.dd-closed {
  border-color: hsl(200,10%,20%);
}

.c2.dd-closed:hover,
.c2.dd-closed:focus {
  border-color: hsl(200,96%,35%);
}

.c2.dd-open {
  border-color: hsl(200,96%,35%);
}

.c4 {
  box-sizing: border-box;
  background-color: hsl(200,10%,20%);
  height: 32px;
  width: 36px;
  border-radius: 1px 20px 20px 1px;
  margin-left: 1px;
  border: 0px;
  outline: none;
  cursor: pointer;
}

.c4 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c4:hover,
.c4:focus {
  background-color: hsl(200,96%,35%);
}

.c4[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 .c3[aria-expanded='true'] {
  background-color: hsl(200,96%,35%);
}

.c0 .c3[aria-expanded='true'] .c5 {
  -webkit-transform: rotate3d(1,0,0,180deg);
  -ms-transform: rotate3d(1,0,0,180deg);
  transform: rotate3d(1,0,0,180deg);
}

.c0 .c3[aria-expanded='true'] + .c1 {
  border-color: hsl(200,96%,35%);
}

<div>
  <div
    class="c0"
  >
    <button
      class="dd-closed c1 c2"
      type="button"
    >
      Test
    </button>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="c3 c4"
      data-reach-menu-button=""
      id="menu-button--1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="c5 c6"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M17.7071068,2.70710678 C18.0976311,2.31658249 18.0976311,1.68341751 17.7071068,1.29289322 C17.3165825,0.902368927 16.6834175,0.902368927 16.2928932,1.29289322 L6.29289322,11.2928932 C5.90236893,11.6834175 5.90236893,12.3165825 6.29289322,12.7071068 L16.2928932,22.7071068 C16.6834175,23.0976311 17.3165825,23.0976311 17.7071068,22.7071068 C18.0976311,22.3165825 18.0976311,21.6834175 17.7071068,21.2928932 L8.41421356,12 L17.7071068,2.70710678 Z"
          transform="rotate(-90 12 12)"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`component: SplitButton with theme customization dropdown should not have box shadows 1`] = `
.c6 {
  font-size: 8px;
  vertical-align: middle;
}

.c2 {
  box-sizing: border-box;
  border: 1px solid;
  border-radius: 20px 1px 1px 20px;
  background-color: hsl(0,0%,100%);
  height: 32px;
  outline: none;
  font-size: 18px;
  line-height: 28px;
  font-weight: 400px;
  cursor: pointer;
  padding-left: 12px;
  padding-right: 12px;
}

.c2 svg {
  margin-right: 4px;
  margin-bottom: 3px;
}

.c2.dd-closed {
  border-color: hsl(200,10%,20%);
}

.c2.dd-closed:hover,
.c2.dd-closed:focus {
  border-color: hsl(200,96%,35%);
}

.c2.dd-open {
  border-color: hsl(200,96%,35%);
}

.c4 {
  box-sizing: border-box;
  background-color: hsl(200,10%,20%);
  height: 32px;
  width: 36px;
  border-radius: 1px 20px 20px 1px;
  margin-left: 1px;
  border: 0px;
  outline: none;
  cursor: pointer;
}

.c4 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c4:hover,
.c4:focus {
  background-color: hsl(200,96%,35%);
}

.c4[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c7 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c7[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c8 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c8[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c9 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c9[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 .c3[aria-expanded='true'] {
  background-color: hsl(200,96%,35%);
}

.c0 .c3[aria-expanded='true'] .c5 {
  -webkit-transform: rotate3d(1,0,0,180deg);
  -ms-transform: rotate3d(1,0,0,180deg);
  transform: rotate3d(1,0,0,180deg);
}

.c0 .c3[aria-expanded='true'] + .c1 {
  border-color: hsl(200,96%,35%);
}

<div>
  <div
    class="c0"
  >
    <button
      class="dd-closed c1 c2"
      type="button"
    >
      Test
    </button>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="c3 c4"
      data-reach-menu-button=""
      id="menu-button--28"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="c5 c6"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M17.7071068,2.70710678 C18.0976311,2.31658249 18.0976311,1.68341751 17.7071068,1.29289322 C17.3165825,0.902368927 16.6834175,0.902368927 16.2928932,1.29289322 L6.29289322,11.2928932 C5.90236893,11.6834175 5.90236893,12.3165825 6.29289322,12.7071068 L16.2928932,22.7071068 C16.6834175,23.0976311 17.3165825,23.0976311 17.7071068,22.7071068 C18.0976311,22.3165825 18.0976311,21.6834175 17.7071068,21.2928932 L8.41421356,12 L17.7071068,2.70710678 Z"
          transform="rotate(-90 12 12)"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`component: SplitButton with theme customization should have 2px borders 1`] = `
.c6 {
  font-size: 8px;
  vertical-align: middle;
}

.c2 {
  box-sizing: border-box;
  border: 2px solid;
  border-radius: 20px 1px 1px 20px;
  background-color: hsl(0,0%,100%);
  height: 32px;
  outline: none;
  font-size: 18px;
  line-height: 28px;
  font-weight: 400px;
  cursor: pointer;
  padding-left: 12px;
  padding-right: 12px;
}

.c2 svg {
  margin-right: 4px;
  margin-bottom: 3px;
}

.c2.dd-closed {
  border-color: hsl(200,10%,20%);
}

.c2.dd-closed:hover,
.c2.dd-closed:focus {
  border-color: hsl(200,96%,35%);
}

.c2.dd-open {
  border-color: hsl(200,96%,35%);
}

.c4 {
  box-sizing: border-box;
  background-color: hsl(200,10%,20%);
  height: 32px;
  width: 36px;
  border-radius: 1px 20px 20px 1px;
  margin-left: 1px;
  border: 0px;
  outline: none;
  cursor: pointer;
}

.c4 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c4:hover,
.c4:focus {
  background-color: hsl(200,96%,35%);
}

.c4[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c7 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c7[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c8 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c8[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c9 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c9[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 .c3[aria-expanded='true'] {
  background-color: hsl(200,96%,35%);
}

.c0 .c3[aria-expanded='true'] .c5 {
  -webkit-transform: rotate3d(1,0,0,180deg);
  -ms-transform: rotate3d(1,0,0,180deg);
  transform: rotate3d(1,0,0,180deg);
}

.c0 .c3[aria-expanded='true'] + .c1 {
  border-color: hsl(200,96%,35%);
}

<div>
  <div
    class="c0"
  >
    <button
      class="dd-closed c1 c2"
      type="button"
    >
      Test
    </button>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="c3 c4"
      data-reach-menu-button=""
      id="menu-button--30"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="c5 c6"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M17.7071068,2.70710678 C18.0976311,2.31658249 18.0976311,1.68341751 17.7071068,1.29289322 C17.3165825,0.902368927 16.6834175,0.902368927 16.2928932,1.29289322 L6.29289322,11.2928932 C5.90236893,11.6834175 5.90236893,12.3165825 6.29289322,12.7071068 L16.2928932,22.7071068 C16.6834175,23.0976311 17.3165825,23.0976311 17.7071068,22.7071068 C18.0976311,22.3165825 18.0976311,21.6834175 17.7071068,21.2928932 L8.41421356,12 L17.7071068,2.70710678 Z"
          transform="rotate(-90 12 12)"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`component: SplitButton with theme customization should have intermediate shape 1`] = `
.c6 {
  font-size: 8px;
  vertical-align: middle;
}

.c2 {
  box-sizing: border-box;
  border: 1px solid;
  border-radius: 8px 1px 1px 8px;
  background-color: hsl(0,0%,100%);
  height: 32px;
  outline: none;
  font-size: 18px;
  line-height: 28px;
  font-weight: 400px;
  cursor: pointer;
  padding-left: 12px;
  padding-right: 12px;
}

.c2 svg {
  margin-right: 4px;
  margin-bottom: 3px;
}

.c2.dd-closed {
  border-color: hsl(200,10%,20%);
}

.c2.dd-closed:hover,
.c2.dd-closed:focus {
  border-color: hsl(200,96%,35%);
}

.c2.dd-open {
  border-color: hsl(200,96%,35%);
}

.c7 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c7[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c8 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c8[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c9 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c9[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c4 {
  box-sizing: border-box;
  background-color: hsl(200,10%,20%);
  height: 32px;
  width: 36px;
  border-radius: 1px 8px 8px 1px;
  margin-left: 1px;
  border: 0px;
  outline: none;
  cursor: pointer;
}

.c4 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c4:hover,
.c4:focus {
  background-color: hsl(200,96%,35%);
}

.c4[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 .c3[aria-expanded='true'] {
  background-color: hsl(200,96%,35%);
}

.c0 .c3[aria-expanded='true'] .c5 {
  -webkit-transform: rotate3d(1,0,0,180deg);
  -ms-transform: rotate3d(1,0,0,180deg);
  transform: rotate3d(1,0,0,180deg);
}

.c0 .c3[aria-expanded='true'] + .c1 {
  border-color: hsl(200,96%,35%);
}

<div>
  <div
    class="c0"
  >
    <button
      class="dd-closed c1 c2"
      type="button"
    >
      Test
    </button>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="c3 c4"
      data-reach-menu-button=""
      id="menu-button--26"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="c5 c6"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M17.7071068,2.70710678 C18.0976311,2.31658249 18.0976311,1.68341751 17.7071068,1.29289322 C17.3165825,0.902368927 16.6834175,0.902368927 16.2928932,1.29289322 L6.29289322,11.2928932 C5.90236893,11.6834175 5.90236893,12.3165825 6.29289322,12.7071068 L16.2928932,22.7071068 C16.6834175,23.0976311 17.3165825,23.0976311 17.7071068,22.7071068 C18.0976311,22.3165825 18.0976311,21.6834175 17.7071068,21.2928932 L8.41421356,12 L17.7071068,2.70710678 Z"
          transform="rotate(-90 12 12)"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`component: SplitButton with theme customization should have square shape 1`] = `
.c6 {
  font-size: 8px;
  vertical-align: middle;
}

.c2 {
  box-sizing: border-box;
  border: 1px solid;
  border-radius: 1px;
  background-color: hsl(0,0%,100%);
  height: 32px;
  outline: none;
  font-size: 18px;
  line-height: 28px;
  font-weight: 400px;
  cursor: pointer;
  padding-left: 12px;
  padding-right: 12px;
}

.c2 svg {
  margin-right: 4px;
  margin-bottom: 3px;
}

.c2.dd-closed {
  border-color: hsl(200,10%,20%);
}

.c2.dd-closed:hover,
.c2.dd-closed:focus {
  border-color: hsl(200,96%,35%);
}

.c2.dd-open {
  border-color: hsl(200,96%,35%);
}

.c7 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c7[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c8 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c8[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c4 {
  box-sizing: border-box;
  background-color: hsl(200,10%,20%);
  height: 32px;
  width: 36px;
  border-radius: 1px;
  margin-left: 1px;
  border: 0px;
  outline: none;
  cursor: pointer;
}

.c4 .c5 {
  width: 10px;
  height: 10px;
  color: hsl(0,0%,100%);
}

.c4:hover,
.c4:focus {
  background-color: hsl(200,96%,35%);
}

.c4[aria-expanded='true'] ~ .c1 {
  border-color: hsl(200,96%,35%);
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 .c3[aria-expanded='true'] {
  background-color: hsl(200,96%,35%);
}

.c0 .c3[aria-expanded='true'] .c5 {
  -webkit-transform: rotate3d(1,0,0,180deg);
  -ms-transform: rotate3d(1,0,0,180deg);
  transform: rotate3d(1,0,0,180deg);
}

.c0 .c3[aria-expanded='true'] + .c1 {
  border-color: hsl(200,96%,35%);
}

<div>
  <div
    class="c0"
  >
    <button
      class="dd-closed c1 c2"
      type="button"
    >
      Test
    </button>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="c3 c4"
      data-reach-menu-button=""
      id="menu-button--24"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="c5 c6"
        fill="currentcolor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M17.7071068,2.70710678 C18.0976311,2.31658249 18.0976311,1.68341751 17.7071068,1.29289322 C17.3165825,0.902368927 16.6834175,0.902368927 16.2928932,1.29289322 L6.29289322,11.2928932 C5.90236893,11.6834175 5.90236893,12.3165825 6.29289322,12.7071068 L16.2928932,22.7071068 C16.6834175,23.0976311 17.3165825,23.0976311 17.7071068,22.7071068 C18.0976311,22.3165825 18.0976311,21.6834175 17.7071068,21.2928932 L8.41421356,12 L17.7071068,2.70710678 Z"
          transform="rotate(-90 12 12)"
        />
      </svg>
    </button>
  </div>
</div>
`;
